### ISSUE TITLE
Undefined index in SandstormSessionProvider

### ISSUE THREAD

**Issue by ocdtrekkie** [zenhack/mediawiki-sandstorm#2]
I don't know if I see anything functionally broken, but the debug log is extremely noisy about this:

`2022/05/14 00:21:45 [error] 13#13: *764 FastCGI sent in stderr: "PHP message: PHP Notice:  Undefined index:  in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 109PHP message: PHP Notice:  Undefined index: sysop in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 114PHP message: PHP Warning:  Use of undefined constant user_id - assumed 'user_id' (this will throw an Error in a future version of PHP) in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 146PHP message: PHP Notice:  Undefined variable: permissions in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 164" while reading response header from upstream, client: 127.0.0.1, server: localhost, request: "GET /index.php?hidebots=1&limit=50&days=7&enhanced=1&title=Special:RecentChanges&urlversion=2&peek=1&from=20220514002118&isAnon=false&action=render HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.3-fpm.sock:", host: "ui-db96a5f88c6e5bd47c8a0527957a5925.ocdhost.sandcats.io:6080"
2022/05/14 00:21:49 [error] 13#13: *766 FastCGI sent in stderr: "PHP message: PHP Notice:  Undefined index:  in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 109PHP message: PHP Notice:  Undefined index: sysop in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 114PHP message: PHP Warning:  Use of undefined constant user_id - assumed 'user_id' (this will throw an Error in a future version of PHP) in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 146PHP message: PHP Notice:  Undefined variable: permissions in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 164" while reading response header from upstream, client: 127.0.0.1, server: localhost, request: "GET /index.php?hidebots=1&limit=50&days=7&enhanced=1&title=Special:RecentChanges&urlversion=2&peek=1&from=20220514002118&isAnon=false&action=render HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.3-fpm.sock:", host: "ui-db96a5f88c6e5bd47c8a0527957a5925.ocdhost.sandcats.io:6080"
2022/05/14 00:21:50 [error] 13#13: *768 FastCGI sent in stderr: "PHP message: PHP Notice:  Undefined index:  in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 109PHP message: PHP Notice:  Undefined index: sysop in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 114PHP message: PHP Warning:  Use of undefined constant user_id - assumed 'user_id' (this will throw an Error in a future version of PHP) in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 146PHP message: PHP Notice:  Undefined variable: permissions in /opt/app/mediawiki/extensions/SandstormAuth/includes/SandstormSessionProvider.php on line 164" while reading response header from upstream,`

**Comment by zenhack** [comment]
While fussing with this I found and fixed a somewhat more serious bug in the auth plugin -- basically, if you made a sharing link with no permissions, then you adding admin privileges retroactively (via the see who has access dialog), then you remove them again, any users which used these links between each step will retain the increased privileges.

Anyway, I just pushed a new spk. It includes the fix for that bug, this, and #1. I forgot to update the changelog, but hopefully this is enough for you to have context.
